homeassistant/components/blink/__init__.py

```diff
@@ -0,0 +1,161 @@
+}
+
+SENSORS = {
+    TYPE_TEMPERATURE: ['Temperature', TEMP_FAHRENHEIT, 'mdi:thermometer'],
+    TYPE_BATTERY: ['Battery', '%', 'mdi:battery-80'],
+    TYPE_WIFI_STRENGTH: ['Wifi Signal', 'dBm', 'mdi:wifi-strength-2'],
+    TYPE_STATUS: ['Status', '', 'mdi:bell']
+}
+
+BINARY_SENSOR_SCHEMA = vol.Schema({
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(BINARY_SENSORS)):
+        vol.All(cv.ensure_list, [vol.In(BINARY_SENSORS)])
+})
+
+SENSOR_SCHEMA = vol.Schema({
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSORS)):
+        vol.All(cv.ensure_list, [vol.In(SENSORS)])
+})
+
+SERVICE_TRIGGER_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string
+})
+
+SERVICE_SAVE_VIDEO_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_FILENAME): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN:
+        vol.Schema({
+            vol.Required(CONF_USERNAME): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+            vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL):
+                cv.time_period,
+            vol.Optional(CONF_BINARY_SENSORS, default={}):
+                BINARY_SENSOR_SCHEMA,
+            vol.Optional(CONF_SENSORS, default={}): SENSOR_SCHEMA,
+        })
+    },
+    extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up Blink System."""
+    from blinkpy import blinkpy
+    conf = config[BLINK_DATA]
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    scan_interval = conf[CONF_SCAN_INTERVAL]
+    hass.data[BLINK_DATA] = blinkpy.Blink(username=username,
+                                          password=password)
+    hass.data[BLINK_DATA].refresh_rate = scan_interval.total_seconds()
+    hass.data[BLINK_DATA].start()
+
+    platforms = [
+        ('alarm_control_panel', {}),
+        ('binary_sensor', conf[CONF_BINARY_SENSORS]),
+        ('camera', {}),
+        ('sensor', conf[CONF_SENSORS]),
+    ]
+
+    for component, schema in platforms:
+        discovery.load_platform(hass, component, DOMAIN, schema, config)
+
+    def trigger_camera(call):
+        """Trigger a camera."""
+        cameras = hass.data[BLINK_DATA].sync.cameras
+        name = call.data[CONF_NAME]
+        if name in cameras:
+            cameras[name].snap_picture()
+        hass.data[BLINK_DATA].refresh(force_cache=True)
+
+    def blink_refresh(event_time):
+        """Call blink to refresh info."""
+        hass.data[BLINK_DATA].refresh(force_cache=True)
+
+    async def async_save_video(call):
+        """Call save video service handler."""
+        await async_handle_save_video_service(hass, call)
+
+    hass.services.register(DOMAIN, SERVICE_REFRESH, blink_refresh)
+    hass.services.register(DOMAIN,
+                           SERVICE_TRIGGER,
+                           trigger_camera,
+                           schema=SERVICE_TRIGGER_SCHEMA)
+    hass.services.async_register(DOMAIN,
```

Comment:
Use the sync version of the register method, not the async one. It doesn't matter that we're registering an async service handler. We're still in a sync context here.